apps/client-asset-sg/src/app/components/disclaimer-dialog/disclaimer-dialog.component.ts

```diff
@@ -1,69 +1,27 @@
-import { MatSelect } from '@angular/material/select';
-import { NavigationEnd, Router } from '@angular/router';
-import { ButtonComponent } from '@asset-sg/client-shared';
-import { isNotNull, isTruthy } from '@asset-sg/core';
-import { Lang } from '@asset-sg/shared';
-import { TranslateModule } from '@ngx-translate/core';
-import { LetModule } from '@rx-angular/template/let';
-import { flow, pipe } from 'fp-ts/function';
-import * as O from 'fp-ts/Option';
-import queryString from 'query-string';
-import { filter, map, startWith } from 'rxjs';
-import { NewlineToBrPipe } from '../../../../../../libs/client-shared/src/lib/pipes/new-lline-to-br.pipe';
-import { LanguageSelectorComponent } from '../language-selector/language-selector.component';
+import { Component, inject } from '@angular/core';
+import { MatDialogRef } from '@angular/material/dialog';
+import { getCurrentLang } from '@asset-sg/client-shared';
+import { TranslateService } from '@ngx-translate/core';
+import { map } from 'rxjs';
+
+const LEGAL_BASE_URL = 'https://www.swissgeol.ch/datenschutz';
 
 @Component({
   selector: 'asset-sg-disclaimer-dialog',
-  standalone: true,
-  imports: [
-    CommonModule,
-    ButtonComponent,
-    FormsModule,
-    MatDialogActions,
-    MatDialogContent,
-    MatError,
-    MatFormField,
-    MatLabel,
-    MatOption,
-    MatSelect,
-    ReactiveFormsModule,
-    TranslateModule,
-    MatDivider,
-    LanguageSelectorComponent,
-    NewlineToBrPipe,
-    LetModule,
-  ],
   templateUrl: './disclaimer-dialog.component.html',
   styleUrl: './disclaimer-dialog.component.scss',
 })
 export class DisclaimerDialogComponent {
-  constructor(private readonly dialogRef: MatDialogRef<DisclaimerDialogComponent>, private readonly router: Router) {}
+  public text = '';
+  private readonly dialogRef = inject(MatDialogRef<DisclaimerDialogComponent>);
+  private readonly translateService = inject(TranslateService);
+
+  public readonly legalUrl$ = getCurrentLang().pipe(
+    map(({ lang }) => (lang === 'de' ? LEGAL_BASE_URL : `${LEGAL_BASE_URL}-${lang}/`))
+  );
 
-  public readonly href = 'https://www.swissgeol.ch/datenschutz';
-  public readonly currentLang$ = this.router.events.pipe(
-    filter((e): e is NavigationEnd => e instanceof NavigationEnd),
-    map((e) => e.urlAfterRedirects),
-    startWith(this.router.url),
-    map(
-      flow(
-        (url) => O.of(url.match('^/(\\w\\w)(.*)$')),
-        O.filter(isTruthy),
-        O.bindTo('match'),
-        O.bind('lang', ({ match }) => pipe(Lang.decode(match[1]), O.fromEither)),
-        O.bind('parsed', ({ match }) => O.of(queryString.parseUrl(match[2]))),
-        O.map(({ lang }) => lang)
-      )
-    ),
-    map((it) => O.toNullable(it)),
-    filter(isNotNull),
-    map((lang) => (lang === 'de' ? this.href : `${this.href}-${lang}/`))
+  public readonly disclaimerText$ = getCurrentLang().pipe(
+    map(() => this.translateService.instant(`disclaimer.content`).replace(/\n/g, '<br>'))
   );
```

Comment:
1. You can `TranslateService.get` over `TranslateService.instant` to get an observable of the translated value that emits if the language changes. This allows you to skip using `getCurrentLang` as base observable.
2. You can replace the regex replacement with a `replaceAll` using the newline character.
```suggestion
  public readonly disclaimerText$ = this.translateService.get(`disclaimer.content`).pipe(
    map((value: string) => value.replaceAll('\n', '<br>'))
  )
```

Reply:
I get an error saying replaceAll does not exist on type String. When follwoing [this](https://stackoverflow.com/questions/63616486/property-replaceall-does-not-exist-on-type-string), a bunch of other error messages appear